packages/report-server/src/reportBuilder/context/buildContext.ts

```diff
@@ -46,21 +51,26 @@ function validateContextProp(key: string): asserts key is keyof typeof contextBu
   }
 }
 
+const setContextValue = <Key extends ContextProp>(
+  context: Context,
+  key: Key,
+  value: Context[Key],
+) => {
+  context[key] = value;
+};
```

Comment:
This is a little trick I've had to employ a few times. Typescript will complain if you do something like:
```
context[key] = await contextBuilders[key](reqContext, data))
```
This is because it can't be sure that `await contextBuilders[key](reqContext, data))` returns a value that's compatible with `context[key]` since it treats `key` as being all of the values for `ContextProp` at once. However, if you use a generic function like this it is able to perform the correct type assertion 👍